server/src/main/java/org/opensearch/common/blobstore/transfer/RemoteTransferContainer.java

```diff
@@ -0,0 +1,252 @@
+ *
+ * @opensearch.internal
+ */
+public class RemoteTransferContainer implements Closeable {
+
+    private int numberOfParts;
+    private long partSize;
+    private long lastPartSize;
+
+    private final long contentLength;
+    private final SetOnce<InputStream[]> inputStreams = new SetOnce<>();
+    private final String fileName;
+    private final String remoteFileName;
+    private final boolean failTransferIfFileExists;
+    private final WritePriority writePriority;
+    private final long expectedChecksum;
+    private final OffsetRangeInputStreamSupplier offsetRangeInputStreamSupplier;
+    private final boolean isRemoteDataIntegritySupported;
+    private final boolean areInputStreamsDecorated;
+
+    private static final Logger log = LogManager.getLogger(RemoteTransferContainer.class);
+
+    /**
+     * Construct a new RemoteTransferContainer object
+     *
+     * @param fileName                       Name of the local file
+     * @param remoteFileName                 Name of the remote file
+     * @param contentLength                  Total content length of the file to be uploaded
+     * @param failTransferIfFileExists       A boolean to determine if upload has to be failed if file exists
+     * @param writePriority                  The {@link WritePriority} of current upload
+     * @param offsetRangeInputStreamSupplier A supplier to create OffsetRangeInputStreams
+     * @param expectedChecksum               The expected checksum value for the file being uploaded. This checksum will be used for local or remote data integrity checks
+     * @param isRemoteDataIntegritySupported A boolean to signify whether the remote repository supports server side data integrity verification
+     * @param areInputStreamsDecorated       A boolean to signify whether the streams created via {@link OffsetRangeInputStreamSupplier#get} are decorated or not
+     */
+    public RemoteTransferContainer(
+        String fileName,
+        String remoteFileName,
+        long contentLength,
+        boolean failTransferIfFileExists,
+        WritePriority writePriority,
+        OffsetRangeInputStreamSupplier offsetRangeInputStreamSupplier,
+        long expectedChecksum,
+        boolean isRemoteDataIntegritySupported,
+        boolean areInputStreamsDecorated
+    ) {
+        this.fileName = fileName;
+        this.remoteFileName = remoteFileName;
+        this.contentLength = contentLength;
+        this.failTransferIfFileExists = failTransferIfFileExists;
+        this.writePriority = writePriority;
+        this.offsetRangeInputStreamSupplier = offsetRangeInputStreamSupplier;
+        this.expectedChecksum = expectedChecksum;
+        this.isRemoteDataIntegritySupported = isRemoteDataIntegritySupported;
+        this.areInputStreamsDecorated = areInputStreamsDecorated;
+    }
+
+    /**
+     * @return The {@link  WriteContext} for the current upload
+     */
+    public WriteContext createWriteContext() {
```

Comment:
Not sure where this is getting consumed, will need some information around this

Reply:
- Segment file path: https://github.com/opensearch-project/OpenSearch/pull/7119/files#diff-f57628264c235a133d2d6db458404e07a43f18026e40ed5e956536c03add9925R445
- Translog file path: https://github.com/opensearch-project/OpenSearch/pull/7119/files#diff-85dadb6157c368957d1d62236381175f24d3f96da4a796947ab22f053646af9aR147